provider/acme/utils.go

```diff
@@ -0,0 +1,68 @@
+package acme
+
+import (
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/containous/flaeg"
+	"github.com/containous/traefik/log"
+	"github.com/xenolf/lego/acme"
+)
+
+func pemEncode(data interface{}) []byte {
```

Comment:
Shouldn't this be moved to the `tls` package instead?

Reply:
I added this test here in the way to avoid error when `acme.json` file is empty, otherwise `acme.jason` hasd to been initialized with `{}`.